### ISSUE TITLE
Python debugging doesn't work in ColabCode (via ngrok and Google Colab)

### ISSUE THREAD

**Issue by krivenkoz** [coder/code-server#2783]
I'm using amazing ColabCode by Abhishek Thakur. It uses ngrok as backbone and Google Colab as "frontend". But when I try to debug Python script, it doesn't launch any debug action. launch.json is empty. When I fill it with next:

"version": "0.2.0",
"configurations": [
    {
        "name": "Python: Current File",
        "type": "python",
        "request": "launch",
        "program": "${file}",
        "console": "integratedTerminal"
    }        

]

I've got next error:

Couldn't find a debug adapter descriptor for debug type 'python' (extension might have failed to activate). Python extension installed properly. Script runs in terminal only (not by F5 button).

How can I debug Python script on ngrok code-server?

**Comment by jsjoeio** [comment]
HI @krivenkoz ! Can you please fill out the issue template?

And can you provide reproduction steps? I'm not super familiar with ColabCode.

And lastly, have you tried debugging a Python script when not using ColabCode? Wondering if there may be an underlying issue.

**Comment by ddurham2** [comment]
I'm seeing the same thing just with code-server 3.9.1 and the latest ms-python extension installed.
creating test.py simply
```
print('hello world')
```
and creating launch.json as
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python: Current File",
            "type": "python",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal"
        }
    ]
}
```

Press F5 in  test.py and it fails with the above error message.  Worse, I don't see any log output anywhere explaining the issue.  Please help.

**Comment by jsjoeio** [comment]
Hey @ddurham2 - thanks for letting us know! Just to clarify, I should be able to reproduce following these steps?

1. install latest version of code-server 3.9.1
2. install the latest ms-python extension
3. create `test.py`
4. crate `launch.json` (using what you have above)
5. hit F5 in `test.py`

## Expected

I don't use Python much these days and also haven't tried debugging it in VS Code/code-server so I'm not sure what should happen?

## Actual

```
Couldn't find a debug adapter descriptor for debug type 'python' (extension might have failed to activate). 
Python extension installed properly. Script runs in terminal only (not by F5 button).
```

Possibly related to: 
- https://github.com/cdr/code-server/issues/2341

**Comment by ddurham2** [comment]
Yes, precisely.  And I think #2341 is what I should focus on, and possibly this issue's fundamental cause too.

I have learned since that if you look in the browser dev console that this behavior is caused by the ms-toolsai.jupyter failing to load.  That seems to be because it's missing some files in the extension ( /home/coder/.local/share/code-server/extensions/ms-toolsai.jupyter-2021.2.0/out/client/* )

I think #2341 already has discovered this.

Also, I've found that ms-python.python@2020.8.101144 is the version just before they started requiring the ms-toolsai.jupyter extension.  I'm currently having to stick with code-server v3.7.1 (due to #2626 )  and specify this version of the ms-python plugin.
